financial-connections/src/main/java/com/stripe/android/financialconnections/repository/ConsumerSessionRepository.kt

```diff
@@ -18,8 +18,11 @@ internal data class CachedConsumerSession(
     val isVerified: Boolean,
 ) : Parcelable
 
-internal interface ConsumerSessionRepository {
+internal fun interface ConsumerSessionProvider {
     fun provideConsumerSession(): CachedConsumerSession?
+}
+
+internal interface ConsumerSessionRepository : ConsumerSessionProvider {
     fun storeConsumerSession(consumerSession: ConsumerSession?)
 }
```

Comment:
Any particular reason to separate these? usually repository's responsibility includes handling reads and writes, and we usually do so on other repos.

Reply:
I think that's ok! just want to make sure we're consistent but better testing is always a good reason 😄